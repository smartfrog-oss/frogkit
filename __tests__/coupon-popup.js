import {
  mount,
  shallow
} from 'vue-test-utils'
import CouponPopup from '@/components/coupon-popup'

const propsData = {
  visible: true,
  config: {
    title: 'Hiergeblieben!',
    headline: 'Smartfrog schenkt dir den ersten Monat!',
    label: 'Gutscheincode:',
    code: 'FREE1',
    accept: 'Jetzt einlÃ¶sen',
    reject: 'Jetzt nicht'
  }

}

describe('Coupon Popup component', () => {
  it('should render component and match snapshot', () => {
    const wrapper = mount(CouponPopup, {
      propsData
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Should emit change event when close method is called', () => {
    const cmp = shallow(CouponPopup, {
      propsData
    })
    const stub = jest.fn()
    cmp.vm.$on('change', stub)
    cmp.vm.close()
    expect(stub).toBeCalled()
  })

  it('Should emit proceed event when proceed method is called', () => {
    const cmp = shallow(CouponPopup, {
      propsData
    })
    const stub = jest.fn()
    cmp.vm.$on('proceed', stub)
    cmp.vm.proceed()
    expect(stub).toBeCalled()
  })
})
