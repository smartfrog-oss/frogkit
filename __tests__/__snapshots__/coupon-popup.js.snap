// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coupon Popup component should render component and match snapshot 1`] = `
<flex justify="center" align="center" class="fk-dialogue v-enter fk-dialogue__animation-in">
  <section class="fk-dialogue__frame"><a class="fk-dialogue__close">
      <icon icon="close-light" color="orange"></icon>
    </a>
    <div class="fk-coupon-popup">
      <title color="primary" size="xl" class="fk-coupon-popup__title">Hiergeblieben!</title>
      <!---->
      <p class="fk-coupon-popup__txt">Smartfrog schenkt dir den ersten Monat!</p>
      <p class="fk-coupon-popup__txt">
        Gutscheincode:
        <span class="fk-coupon-popup__txt__code">FREE1</span></p> <button size="big" block="true" color="secondary" link="" class="fk-coupon-popup__btn"><span>Jetzt einl√∂sen</span>
        <icon icon="arrow-right"></icon></button> <span class="fk-coupon-popup__close">Jetzt nicht</span>
    </div>
  </section>
</flex>
`;
